serializer-configurate3/src/main/java/net/kyori/adventure/serializer/configurate3/ConfigurateComponentSerializer.java

```diff
@@ -0,0 +1,107 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.serializer.configurate3;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.serializer.ComponentSerializer;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.objectmapping.serialize.TypeSerializer;
+import ninja.leaping.configurate.objectmapping.serialize.TypeSerializerCollection;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * A serializer that will output to Configurate {@link ConfigurationNode}s.
+ *
+ * <p>This serializer only modifies its own serializer collection. To add to another collection, 
+ * use an existing serializer instance to {@linkplain #populate(TypeSerializerCollection) populate} 
+ * that collection. Serializers are added for every Adventure data type.</p>
+ *
+ * <p>The preferred way to use Configurate integration is by registering this serializer's
+ * {@linkplain TypeSerializer type serializers} with a separate </p>
+ */
+public interface ConfigurateComponentSerializer extends ComponentSerializer<Component, Component, ConfigurationNode> {
+  /**
+   * Create a new builder for a customized Configurate serializer
+   *
+   * @return a new builder
+   */
+  static @NonNull Builder builder() {
+    return new ConfigurateComponentSerializerImpl.Builder();
+  }
+
+  /**
+   * Get an instance with default settings.
+   *
+   * @return the shared default instance
+   */
+  static @NonNull ConfigurateComponentSerializer configurate() {
+    return ConfigurateComponentSerializerImpl.INSTANCE;
+  }
+
+  // TODO @ Configurate 4: expose serializer collection for new child creation
+
+  /**
+   * Populate an existing serializer collection with Adventure serializers.
+   *
+   * @param collection collection to populate
+   * @return input collection
+   */
+  @NonNull TypeSerializerCollection populate(final @NonNull TypeSerializerCollection collection);
```

Comment:
Is there a better name we can find for this method?

Reply:
populate is what i've used in the past for this -- alternative would be `addSerializersTo` or something?

Reply:
It is kinda ugly, but I do like it better in this case I think.